### DISCUSSION TITLE
How to change the schema of references property in the JS API?

### DISCUSSION THREAD

**Discussion by ErikKoinberg** [bhch/django-jsonform#153] [Q&A] [answered]
I have an object that has a property that is defined inside $defs and linked to using $ref, like so: 
`
'type': 'object',
'properties': {
'some_property': {'$ref': '#/$defs/property_definition'}
}
`
When trying to change something about the schema using the JS API, I am unable to do so, because the e.schema dict does not recognise that the ref points somewhere. For instance, I might want to make the some_property hidden using `'widget': 'hidden'`. How can that be done?

**Comment by ErikKoinberg** [comment]
I suppose that unless the defs are compiled into individual properties, it is not going to be possible to change an aspect of a single referencing instance, and changing it in the defs is also not an options since that would change other input firlds as well. I guess that i have to make them individual, but perhaps putting shared code in Python dicts that are then referenced to in the main schema.

**Reply by bhch** [reply, marked as answer]
> I suppose that unless the defs are compiled into individual properties, it is not going to be possible to change an aspect of a single referencing instance

Overriding specific properties of a referenced definition is supported.

You can paste this schema in the [playground](https://bhch.github.io/react-json-form/playground/) and the `last_name` field should be hidden:

```json
{
  "type": "object",
  "properties": {
    "first_name": {
      "$ref": "#/$defs/name"
    },
    "last_name": {
      "$ref": "#/$defs/name",
      "widget": "hidden"
    }
  },
  "$defs": {
    "name": {"type": "string"}
  }
}
```

**Comment by bhch** [comment]
Can you give me some code to be able to reproduce it?
